validator-api/src/main.rs

```diff
@@ -267,6 +276,45 @@ async fn get_gateways(cache: &State<Arc<ValidatorCache>>) -> Json<Cache<Vec<Gate
     Json(cache.gateways().await)
 }
 
+//  All strings are base58 encoded representations of structs
+#[derive(Deserialize, CopyGetters)]
+struct BlindSignRequestBody {
```

Comment:
Could you move those to the `coconut` specific module? I've cleaned up the `main.rs` significantly in my PR that will come soon ™️   (https://github.com/nymtech/nym/pull/680). You could also move the routes away and perhaps create a full Fairing (https://rocket.rs/v0.5-rc/guide/fairings/#fairings).
Take a look at what I've done in my PR. I've modified your validator cache to work that way (and if you disagree with my decisions, even better! We could discuss that)